### DISCUSSION TITLE
Does queryClient.fetchInfiniteQuery refetch previous pages?

### DISCUSSION THREAD

**Discussion by kimurakenshi** [TanStack/query#5548] [Q&A] [answered]
Will this code always fetch the first page and remove any previous pages from the cache or will refetch everything?

```ts
const data = await queryClient.fetchInfiniteQuery({ queryKey, queryFn })
```

Should I use ` queryClient.removeQueries` to clear the cache before fetching?

My use case for this is a grid that has infinite scrolling. After user scrolls a few times, new pages will loaded but if a filter is applied, the infinite query needs to start over, clear the query cache and only fetch the first page.

**Comment by TkDodo** [comment, marked as answer]
filter needs to be part of the queryKey then.
